pystiche/papers/sanakoyeu_et_al_2018/data.py

```diff
@@ -0,0 +1,159 @@
+from typing import Optional, Sized, Tuple, List, Iterator, Dict, Any, Union
+import torch
+from torch.utils.data import Dataset, Sampler, DataLoader
+from pystiche.data import (
+    ImageFolderDataset,
+    FiniteCycleBatchSampler,
+    InfiniteCycleBatchSampler,
+)
+from pystiche.image.transforms import (
+    Transform,
+    ComposedTransform,
+    ValidRandomCrop,
+)
+from pystiche.image import extract_num_channels
+from pystiche.image.transforms.functional import grayscale_to_fakegrayscale
+from pystiche.image.transforms import functional as F
+
+
+class FiniteChangingCycleBatchSampler(InfiniteCycleBatchSampler):
```

Comment:
This `BatchSampler` is the idea of me to implement a changing number of `num_batches` during the training. The author first trains for `2e5` batches and then trains with a reduced `lr` for further `1e5` batches.

Reply:
In that case I think it would be more clear to simply implement a custom training loop. You can keep the `lr_scheduler` as the learning rate also changes, but I would recreate the `batch_sampler` for every epoch.